pkg/commands/git.go

```diff
@@ -163,7 +163,7 @@ func NewGitCommandAux(
 		Bisect:      bisectCommands,
 		WorkingTree: workingTreeCommands,
 		Worktree:    worktreeCommands,
-		Version:     version,
+		Version:     repoPathCache.GetGitVersion(),
```

Comment:
exposing `GetGitVersion` on repoPathCache is a little too opportunistic for my liking: the fact that repo paths knows about the git version doesn't mean we should rely on it for obtaining the git version. I would instead pass that through to NewGitCommand explicitly as we did before.

Reply:
Good catch. To be honest, I found a bit of "architectural pull" towards something which is really a "Git service", but explicitly tried to avoid in this PR(\*\*). Looks like some of that slipped through here. 

\*\*: Use cases that motivate that are things like [this comment from Stefan](https://github.com/jesseduffield/lazygit/issues/3306#issuecomment-1926569061), specifically needing to cd into the repo consistently as part of a git operation. One could envision a first class GitRepo entity (basically what we know as RepoPaths), but with an upgraded API to consolidate repo handling logic. The code logic shifts to asking "perform this git operation in this Repo".  The cache is essentially an implementation detail of the repo accessor API at that point. I'm specifically not saying we *should* go there at this time, just that it's something to chew on if/when more use cases accumulate that support such a direction.

Reply:
That comment was specific to those few submodule operations though, and those can be dealt with locally in `pkg/commands/git_commands/submodule.go` (I'll have some code to look at shortly). I don't see where else we have the need to perform a git operation in a repo other than the currently open one.